aws/resource_aws_servicecatalog_provisioned_product.go

```diff
@@ -0,0 +1,719 @@
+			},
+			"arn": { //o=ppd
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"cloudwatch_dashboard_names": { //o=ppd
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"created_time": { //o=ppd,o=rd
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"ignore_errors": { //d
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"last_provisioning_record_id": { //o=ppd
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"last_record_id": { //o=ppd
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"last_successful_provisioning_record_id": { //o=ppd
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"launch_role_arn": { //o=ppd,o=rd
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": { //io=ppd,o=rd
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"notification_arns": { //i
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"path_id": { //iuo=rd
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ConflictsWith: []string{
+					"path_name",
+				},
+			},
+			"path_name": { //iu
+				Type:     schema.TypeString,
+				Optional: true,
+				ConflictsWith: []string{
+					"path_id",
+				},
+			},
+			"product_id": { //iuo=ppd,o=rd
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ExactlyOneOf: []string{
+					"product_id",
+					"product_name",
+				},
+			},
+			"product_name": { //iu
+				Type:     schema.TypeString,
+				Optional: true,
+				ExactlyOneOf: []string{
+					"product_id",
+					"product_name",
+				},
+			},
+			"provisioning_artifact_id": { //iuo=ppd,o=rd
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ExactlyOneOf: []string{
+					"provisioning_artifact_id",
+					"provisioning_artifact_name",
+				},
+			},
+			"provisioning_artifact_name": { //iu
+				Type:     schema.TypeString,
+				Optional: true,
+				ExactlyOneOf: []string{
+					"provisioning_artifact_id",
+					"provisioning_artifact_name",
+				},
+			},
+			"provisioning_parameters": { //iu
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"key": { //iu
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"use_previous_value": { //u
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"value": { //iu
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"provisioning_preferences": { //iu
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"accounts": { //iu
```

Comment:
The underlying API field for this (and other attributes in this block) is names `StackSetAccount` - Consider renaming to `stack_set_account`?

Reply:
Instead of rename each of the arguments in the `provisioning_preferences` block, with seems redundant, repeated, and repetitive, I renamed the block to `stack_set_provisioning_preferences`. What do you think?